server/sonar-process-monitor/src/main/java/org/sonar/application/process/ProcessLauncherImpl.java

```diff
@@ -0,0 +1,256 @@
+  public void close() {
+    allProcessesCommands.close();
+  }
+
+  @Override
+  public ProcessMonitor launch(EsCommand esCommand) {
+    Process process = null;
+    try {
+      writeConfFiles(esCommand);
+      ProcessBuilder processBuilder = create(esCommand);
+      LOG.info("Launch process[{}]: {}", esCommand.getProcessId().getKey(), String.join(" ", processBuilder.command()));
+
+      process = processBuilder.start();
+
+      return new EsProcessMonitor(process, esCommand);
+    } catch (Exception e) {
+      // just in case
+      if (process != null) {
+        process.destroyForcibly();
+      }
+      throw new IllegalStateException(format("Fail to launch process [%s]", esCommand.getProcessId().getKey()), e);
+    }
+  }
+
+  private void writeConfFiles(EsCommand esCommand) {
+    File confDir = esCommand.getConfDir();
+    if (!confDir.exists() && !confDir.mkdirs()) {
+      String error = format("Failed to create temporary configuration directory [%s]", confDir.getAbsolutePath());
+      LOG.error(error);
+      throw new IllegalStateException(error);
+    }
+
+    try {
+      IOUtils.copy(getClass().getResourceAsStream("elasticsearch.yml"), new FileOutputStream(new File(confDir, "elasticsearch.yml")));
+      IOUtils.copy(getClass().getResourceAsStream("jvm.options"), new FileOutputStream(new File(confDir, "jvm.options")));
+      esCommand.getLog4j2Properties().store(new FileOutputStream(new File(confDir, "log4j2.properties")), "log42 properties file for ES bundled in SonarQube");
+    } catch (IOException e) {
+      e.printStackTrace();
+      throw new IllegalStateException("Failed to write ES configuration files", e);
+    }
+  }
+
+  @Override
+  public ProcessMonitor launch(JavaCommand javaCommand) {
+    Process process = null;
+    try {
+      ProcessCommands commands = allProcessesCommands.createAfterClean(javaCommand.getProcessId().getIpcIndex());
+
+      ProcessBuilder processBuilder = create(javaCommand);
+      LOG.info("Launch process[{}]: {}", javaCommand.getProcessId().getKey(), String.join(" ", processBuilder.command()));
+      process = processBuilder.start();
+      return new ProcessCommandsProcessMonitor(process, commands);
+    } catch (Exception e) {
+      // just in case
+      if (process != null) {
+        process.destroyForcibly();
+      }
+      throw new IllegalStateException(format("Fail to launch process [%s]", javaCommand.getProcessId().getKey()), e);
+    }
+  }
+
+  private ProcessBuilder create(EsCommand esCommand) {
+    List<String> commands = new ArrayList<>();
+    commands.add(esCommand.getExecutable().getAbsolutePath());
+    commands.addAll(esCommand.getEsOptions());
+
+    return create(esCommand, commands);
+  }
+
+  private ProcessBuilder create(JavaCommand javaCommand) {
+    List<String> commands = new ArrayList<>();
+    commands.add(buildJavaPath());
+    commands.addAll(javaCommand.getJavaOptions());
+    // TODO warning - does it work if temp dir contains a whitespace ?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)